src/openrct2/localisation/LanguagePack.cpp

```diff
@@ -621,9 +619,9 @@ class LanguagePack final : public ILanguagePack
         if (_currentGroup == nullptr)
         {
             // Make sure the list is big enough to contain this string id
-            while (_strings.size() <= (size_t)stringId)
+            if ((size_t)stringId >= _strings.size())
             {
-                _strings.push_back(nullptr);
+                _strings.resize(stringId + 1);
```

Comment:
Does `resize` guarantee `nullptr`?

Reply:
It creates a default value, so in this case it is `nullptr`, yes.

Reply:
`resize` defaults initializes, `reserve` does not.